src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/AzureFunction/InvokeFunctionAsyncWrapper.cs

```diff
@@ -0,0 +1,202 @@
+// Copyright 2020 New Relic, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Collections;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using NewRelic.Agent.Api;
+using NewRelic.Agent.Extensions.Providers.Wrapper;
+using NewRelic.Reflection;
+
+namespace NewRelic.Providers.Wrapper.AzureFunction
+{
+    public class InvokeFunctionAsyncWrapper : IWrapper
+    {
+        private static bool _loggedDisabledMessage;
+        private const string WrapperName = "AzureFunctionInvokeAsyncWrapper";
+
+        private static bool _coldStart = true;
+        private static bool IsColdStart => _coldStart && !(_coldStart = false);
+
+        public bool IsTransactionRequired => false;
+
+        public CanWrapResponse CanWrap(InstrumentedMethodInfo methodInfo)
+        {
+            return new CanWrapResponse(WrapperName.Equals(methodInfo.RequestedWrapperName));
+        }
+
+        public AfterWrappedMethodDelegate BeforeWrappedMethod(InstrumentedMethodCall instrumentedMethodCall, IAgent agent,
+            ITransaction transaction)
+        {
+            if (!agent.Configuration.AzureFunctionModeEnabled) // bail early if azure function mode isn't enabled
+            {
+                if (!_loggedDisabledMessage)
+                {
+                    agent.Logger.Info("Azure Function mode is not enabled; Azure Functions will not be instrumented.");
+                    _loggedDisabledMessage = true;
+                }
+
+                return Delegates.NoOp;
+            }
+
+            dynamic functionContext = instrumentedMethodCall.MethodCall.MethodArguments[0];
+
+            if (functionContext == null)
+            {
+                agent.Logger.Debug($"{WrapperName}: FunctionContext is null, can't instrument this invocation.");
+                throw new ArgumentNullException("functionContext");
+            }
+
+            var functionDetails = new FunctionDetails(functionContext, agent);
+            if (!functionDetails.IsValid())
+            {
+                agent.Logger.Debug($"{WrapperName}: FunctionDetails are invalid, can't instrument this invocation.");
+                throw new Exception("FunctionDetails are missing some require information.");
+            }
+
+            transaction = agent.CreateTransaction(
+                isWeb: functionDetails.IsWebTrigger,
+                category: "AzureFunction",
+                transactionDisplayName: functionDetails.FunctionName,
+                doNotTrackAsUnitOfWork: true);
+
+            if (instrumentedMethodCall.IsAsync)
+            {
+                transaction.AttachToAsync();
+                transaction.DetachFromPrimary(); //Remove from thread-local type storage
+            }
+
+            if (IsColdStart) // only report this attribute if it's a cold start
+            {
+                transaction.AddFaasAttribute("faas.coldStart", "true");
```

Comment:
The OTel conventions have this as a boolean instead of a string. Should we make this a boolean too, even though the equivalent lambda value is a string?